### PR TITLE
Rh/auth and invoice

### PR DESCRIPTION
**Pull request by Gambarou**
### Briefly describe your changes:
Added authentication for previously public routers (invoice/[invoiceId] and api/invoice) using JWT to encode the invoiceId we receive in the payment link.
Changed the invoice link structure from query params to now a dynamic route.
Added some more functionality to the invoice page, such that when an invoice status is anything but open users will not be able to click the pay invoice link. Also shows the status of the invoice with a badge. 

**Jira issue number and link:**
https://pltg.atlassian.net/browse/PLTG-41?atlOrigin=eyJpIjoiMjM2NmEwMGVkOWVmNGMxMTlmZDRhYmQ2MDM5N2I0ODYiLCJwIjoiaiJ9

PLTG-41
**Type of change:**
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### Please answer the following:

**1. What is the current behavior?**
*If the feature did not exist, please say so.*
Routes were previously open, which was a security issue.
**2. What is the new behavior?**
The routes are now secured using JWT.
**3. What tests did you run?**

### Checklist before requesting a review
- [x] I have performed a self-review of my code.
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests for my code.
- [x] I have resolved all merge conflicts.

### CODE REVIEW

File: bill-bot-baggins/components/Header.tsx
Comment (cchrty) [inline bill-bot-baggins/components/Header.tsx:1]: refactor to remove nested navbar component if possible

File: bill-bot-baggins/components/Profile.tsx
Comment (cchrty) [inline bill-bot-baggins/components/Profile.tsx:1]: may need to be updated - client no longer logging in/out

**Review by cchrty — APPROVED** [review]
Great job with auth!

**Review by lhodges3 — APPROVED** [review]
Looks job Robert, looks great!

**Pull request merged**